lib/ex_unit/lib/ex_unit/callbacks.ex

```diff
@@ -612,6 +612,19 @@ defmodule ExUnit.Callbacks do
 
   Note that if the started process terminates before it is linked to the test process,
   this function will exit with reason `:noproc`.
+
+  > #### To Link or Not To Link {: .warning}
+  >
+  > When using `start_link_supervised!/2`, be aware that the order of process termination
+  > during test exit is not guaranteed (for the processes started with
+  > `start_link_supervised!/2`). When the test process exits, the crash signal propagates
+  > to all linked processes virtually simultaneously, which can lead to processes
+  > terminating in an unpredictable order. This is particularly problematic when you have
+  > processes that the test starts with `start_link_supervised!/2` and that depend  on
+  > each other.
+  >
+  > If you need guaranteed shutdown order, use `start_supervised/2`. With that, processes
+  > are shut down *by the test supervisor* in reverse order, ensuring graceful termination.
```

Comment:
I don't think the order is not guaranteed, it is more that the order can be interrupted because of links. It is a nitpick but i think it is an important difference in semantics. I also don't think we titlecase titles:

```suggestion
  > #### To link or not to link {: .warning}
  >
  > When using `start_link_supervised!/2`, the test process will be linked to the
  > spawned processes. When the test process exits, it exits with reason `:shutdown`,
  > and the crash signal propagates to all linked processes virtually simultaneously,
  > which can lead to processes terminating in an unpredictable order if they are not
  > trapping exits. This is particularly problematic when you have processes that the
  > test starts with `start_link_supervised!/2` and that depend on each other.
  >
  > If you need guaranteed shutdown order, use `start_supervised!/2`. This way the
  > test process exiting does not affect the started processes, and they will be shut down
  > *by the test supervisor* in reverse order, ensuring graceful termination.
```

WDYT?